WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -295,30 +295,55 @@ def do_actions_for_selected
     @show_events = params[:show_events] == "true"
     @competition = competition_from_params
     registrations = @competition.registrations.find(selected_registrations_ids)
+    count_success = 0
+    registration_error_ids = []
 
     case params[:registrations_action]
     when "accept-selected"
       registrations.each do |registration|
         if !registration.accepted?
-          registration.update!(accepted_at: Time.now, accepted_by: current_user.id, deleted_at: nil)
-          RegistrationsMailer.notify_registrant_of_accepted_registration(registration).deliver_later
+          if registration.update(accepted_at: Time.now, accepted_by: current_user.id, deleted_at: nil)
+            count_success += 1
+            RegistrationsMailer.notify_registrant_of_accepted_registration(registration).deliver_later
+          else
+            registration_error_ids << registration.id
+            flash.now[:danger] = registration.errors.full_messages.join(', ')
```

Comment:
Looks like this overrides the flash message for any failed registration update, it would make sense to show all the errors. One idea would be keeping an array with errors and adding something along the lines of "Failed to update <registration name>: <error messages>" and then join them all.

Reply:
Done!